src/stores/RoomViewStore.tsx

```diff
@@ -337,6 +337,17 @@ class RoomViewStore extends Store<ActionPayload> {
                     msg = _t("The person who invited you already left the room, or their server is offline.");
                 }
             }
+
+            // if joined through room ID and didn't provide any via servers, provide
+            // a more detailed error than "No known servers"
+            if (roomId === this.state.roomId && this.state.viaServers.length === 0) {
```

Comment:
```suggestion
            if (roomId === this.state.roomId && !this.state.viaServers?.length) {
```

viaServers is nullable, typing in this store type is lacking